src/main/java/model/CodeQualityRule.java

```diff
@@ -0,0 +1,68 @@
+package main.java.model;
+
+/**
+ * A class that represents a rule based on 
+ * predetermined or user-defined metrics
+ * and thresholds to detect code defects
+ * and thus determine code quality
+ * 
+ * @author Lino Silva
+ */
+public class CodeQualityRule {
+
+	private String name;
+	private boolean isDefault;
+	private String rule;

```

Comment:
As we discussed, let's add a `isBasicRule` boolean to store if the rule is basic or advanced.

Reply:
maybe call it isAdvancedGuiRule to be more self-explanatory

Reply:
maybe call it isAdvancedGuiRule to be more self-explanatory

Reply:
I don't think we should name model attributes based on how it is going to be presented in the ui. The model is independent of how any ui is going to display it. If this name is not clear, we either add javadocs explaining it (that's actually something we should have been doing already now that I think about, let's add a ticket for it) or think of a different name.

Reply:
the rule is basic or advanced *depending* on the UI presented to the user, correct? because it's saved in a different way depending on how it was presented in the GUI. I don't see a problem in naming it according to the way it was generated. @lrnsa-iscteiul what do you think?

Reply:
I think it should be named just isAdvanced